python/beeai_framework/tools/search/wikipedia.py

```diff
@@ -13,16 +13,54 @@
 # limitations under the License.
 
 
+from typing import Any
+
+import wikipediaapi
+from pydantic import BaseModel, Field
+
+from beeai_framework.tools import ToolError
+from beeai_framework.tools.search import SearchToolOutput, SearchToolResult
 from beeai_framework.tools.tool import Tool
 
 
-class WikipediaTool(Tool):
+class WikipediaSearchToolInput(BaseModel):
+    query: str = Field(description="Search query, name of the Wikipedia page.")
+
+
+class WikipediaSearchToolResult(SearchToolResult):
+    pass
+
+
+class WikipediaSearchToolOutput(SearchToolOutput):
+    pass
+
+
+class WikipediaSearchTool(Tool[WikipediaSearchToolInput]):
     name = "Wikipedia"
-    description = "Search factual and historical information, including biography, history, politics, geography, society, culture, science, technology, people, animal species, mathematics, and other subjects."  # noqa: E501
+    description = "Search factual and historical information, including biography, \
+        history, politics, geography, society, culture, science, technology, people, \
+        animal species, mathematics, and other subjects."
+    input_schema = WikipediaSearchToolInput
+    client = wikipediaapi.Wikipedia(
+        user_agent="beeai-framework https://github.com/i-am-bee/beeai-framework", language="en"
+    )
 
-    def input_schema(self) -> str:
-        # TODO: remove hard code
-        return '{"type":"object","properties":{"query":{"type":"string","format":"date","description":"Name of the wikipedia page, for example \'New York\'"}}}'  # noqa: E501
+    def __init__(self, full_text: bool = False) -> None:
+        super().__init__()
+        self.full_text = full_text
```

Comment:
Use `BaseModel` object instead (`WikipediaInput` / `WikipediaOptions`) to prevent further breaking changes.
Add more options that the client provides.